src/shogun/base/SGObject.h

```diff
@@ -1081,6 +1078,20 @@ class CSGObject
 	SGSubscriber* m_subscriber_params;
 };
 
+template <class T>
+T* make_clone(T* orig, ParameterProperties pp = ParameterProperties::ALL)
+{
+	REQUIRE(orig, "No object provided.\n");
+	return orig->clone(pp)->template as<T>();
+}
+
+template <class T>
+const T* make_clone(const T* orig, ParameterProperties pp = ParameterProperties::ALL)
```

Comment:
so I can clone and get back the same type as I provided, it is annoying to always have to cast the result

Reply:
const verison is required from xvalidation